src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -40,46 +40,43 @@ interface FormFieldParam {
   placeholderText: string
   errorMessage: string
 }
-interface ImplicitFormFieldParam<T> {
-  value: T
+interface ImplicitParam<T, K extends keyof T> {
+  name: string
+  value: T[K]
 }
 
-interface SchemaMetadata {
-  [FiatAccountSchema.AccountNumber]: {
-    [Property in keyof AccountNumber]:
-      | FormFieldParam
-      | ImplicitFormFieldParam<AccountNumber[Property]>
-  }
+type AccountNumberSchema = {
+  [Property in keyof AccountNumber]: FormFieldParam | ImplicitParam<AccountNumber, Property>
 }
 
-// This is a mapping between different fiat account schema to the metadata of the fields that need to be rendered on the bank details screen
-const SCHEMA_TO_FIELD_METADATA_MAP: SchemaMetadata = {
-  [FiatAccountSchema.AccountNumber]: {
-    accountName: {
-      name: 'accountName',
-      label: i18n.t('fiatAccountSchema.accountName.label'),
-      regex: /.*?/,
-      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
-      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
-    },
-    institutionName: {
-      name: 'institutionName',
-      label: i18n.t('fiatAccountSchema.institutionName.label'),
-      regex: /.*?/,
-      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
-      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
-    },
-    accountNumber: {
-      name: 'accountNumber',
-      label: i18n.t('fiatAccountSchema.accountNumber.label'),
-      regex: /^[0-9]{10}$/,
-      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
-      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
-    },
-    country: { value: 'US' },
-    fiatAccountType: { value: FiatAccountType.BankAccount },
+const getAccountNumberSchema = (implicitParams: {
```

Comment:
Moving the data from a const with all the schemas to a function that returns a single schema is beneficial because different schemas may have different implicitParams.

Reply:
agree with this as well